### DISCUSSION TITLE
New function to transform Exposure to Spending

### DISCUSSION THREAD

**Discussion by Crypto1993** [facebookexperimental/Robyn#757] [Ideas]
Hello everyone! 
I saw from different discussions  and sourcecode that robyn change its modeling aproach from using exposure variables directly to spending. The reason was that Mic-Men function doesn't fit well exposure to spending and therefore the allocator suffers from impresition. Since fitting directly Exposure would provide a better model that is not effected by seasonality, discounts, varying prices for media, dow you think it would be a good a idea to introduce a function that receives a vector of spot prices for media exposure over the date range to optimize? 
Thank you

**Comment by gufengzhou** [comment]
Sorry for the late reply. The "spot price" method was in use before and is the avg price for the model period, and it was fitting badly for many cases, leading to terrible budget allocation. We're aware of the arguments for exposure fitting. But we haven't found a reliable way to translate exposure to spend sofar. 

In fact, there's a study indicating no significant difference between exposure vs spend fitting an MMM. It makes sense to me, because MMM depends on time series variation and the difference between spend and imp is really not big in most cases. The fine-grained price fluctuations and seasonality etc that you're assuming are not really reflected on the aggregate of time series. 

We're open integrate to a more sophisticated solution and re-introduce the exposure if available. 

http://business-science.pl/dont-get-the-wrong-impression-assessing-the-best-input-variable-to-reflect-meta-in-mmm/
